PackagingCon2021-fpm/main.tex

```diff
@@ -0,0 +1,320 @@
+   \end{columns}
+   \vskip1ex
+
+   \begin{columns}
+      \hskip.1\textwidth
+      \begin{column}{.6\textwidth}
+         \begin{itemize}
+            \item creates new projects with fpm layout
+            \item backfilling to update existing projects
+         \end{itemize}
+      \end{column}
+   \end{columns}
+\end{frame}
+
+\begin{frame}{Layout of an fpm project}
+   \begin{columns}
+      \begin{column}{.6\textwidth}
+         \includegraphics[width=\textwidth]{Figures/layout.pdf}
+      \end{column}
+      \hskip-1em
+      \begin{column}{.45\textwidth}
+         \begin{itemize}
+            \item package manifest in \alert{TOML}
+            \item limited complexity of build file
+            \item mainly meta data of project\\
+               (license, author, keywords, \ldots)
+               \\[2ex]
+            \item default layout requires only \alert{name}\\
+               (\textcolor{blue}{src}, \textcolor{blue}{app}, \textcolor{blue}{example}, and \textcolor{blue}{test})
+            \item \alert{source-dir} for customizing layout
+               \\[2ex]
+            \item executable names from \alert{program} unit
+               \\[2ex]
+            \item system libraries can be linked
+               \\[2ex]
+         \end{itemize}
+      \end{column}
+   \end{columns}
+\end{frame}
+
+\begin{frame}{First-class dependencies}
+   \begin{itemize}
+      \item dependencies are currently specified by referencing another project
+         (\textit{e.\,g.} via git URL)
+   \end{itemize}
+
+   \begin{center}
+      \includegraphics[width=.85\textwidth]{Figures/dependencies.pdf}
+   \end{center}
+
+   \begin{itemize}
+      \item different scopes available beside full dependencies
+      \item can depend on projects only for testing or limit dependencies to single executable
+   \end{itemize}
+\end{frame}
+
+\begin{frame}{Extending fpm with plugins}
+   \begin{itemize}
+      \item package manifest provides space for third-party projects
+      \item fpm is available as fpm package and can be used as dependency
+      \item plugins are used automatically from the command-line
+      \item useful for staging new features before integration in fpm
+   \end{itemize}
+   \vfill
+
+   \begin{columns}
+      \begin{column}{.45\textwidth}
+         Searching for fpm packages: \alert{fpm-search}
+         \begin{itemize}
+            \item available from \textit{fpm search} command
+            \item downloads the fpm registry
+            \item search package names, descriptions
+         \end{itemize}
+      \end{column}
+      \hskip1em
+      \begin{column}{.45\textwidth}
+         Documentation of intrinsics: \alert{fpm-man}
+         \begin{itemize}
+            \item available from \textit{fpm man} command
+            \item access to documentation of intrinsics
+            \item cross-platform, no man required
+         \end{itemize}
+      \end{column}
+   \end{columns}
+\end{frame}
+
+
+\section{Summary and outlook}
+
```

Comment:
Should we have a Next steps slide? Listing some near-future milestones for people to get excited about.

Reply:
Fpm is a rather young project and I actually want to motivate the discussion to get them input to steer us into the future. That's why I went with the open questions rather than milestones here.